cpp/tests/join/semi_anti_join_tests.cpp

```diff
@@ -380,3 +382,36 @@ TEST_F(JoinTest, SemiJoinEmptyTables)
     CUDF_TEST_EXPECT_TABLES_EQUIVALENT(*expected, *result);
   }
 }
+
+TEST_F(JoinTest, AntiSemiJoinLargeExtentOverflowPrevention)
+{
+  // Test validates size_t extent can handle bucket storage sizes that would
+  // overflow int32_t extent when compute_bucket_storage_size() uses low load factors
+  constexpr cudf::size_type table_size = 10000000;  // 10M rows
+  constexpr double load_factor         = 0.0025;    // Hash table extent would be ~4B
+
+  // Compile-time validation
+  constexpr auto expected_bucket_size = static_cast<size_t>(table_size / load_factor);
+  static_assert(expected_bucket_size > std::numeric_limits<cudf::size_type>::max(),
+                "Bucket storage size should be significant");
+
+  // Create test tables and validate semi join operations succeed
+  auto const init = cudf::numeric_scalar<cudf::size_type>{0};
+  auto build_col  = cudf::sequence(table_size, init, cudf::numeric_scalar<cudf::size_type>{1});
+
+  auto build_table = cudf::table_view{{build_col->view()}};
+  cudf::table empty_probe_table{};
+
+  // Test with load factors that would cause overflow in int32_t extent
```

Comment:
Can you run a compute-sanitizer on this test?
Just concerned on how much time this will cost.